pkg/observer/monitor/monitor.go

```diff
@@ -45,27 +38,32 @@ type ClientProviders struct {
 	Blockchain common.BlockchainClientProvider
 }
 
+type metaDataStore interface {
```

Comment:
I was going to say that I usually see metadata as a single word, but I see that it was snake-cased previously.